built_value_generator/lib/src/serializer_source_class.dart

```diff
@@ -304,25 +304,25 @@ abstract class SerializerSourceClass
   String generateSerializer() {
     if (isBuiltValue) {
       return '''
-class ${serializerImplName} implements StructuredSerializer<$genericName> {
+class $serializerImplName implements StructuredSerializer<$genericName> {
   @override
   final Iterable<Type> types = const [$name, $implName];
   @override
   final String wireName = '${escapeString(wireName)}';
 
   @override
-  Iterable<Object> serialize(Serializers serializers, $genericName object,
+  Iterable<Object$orNull> serialize(Serializers serializers, $genericName object,
       {FullType specifiedType = FullType.unspecified}) {
-    ${fields.isEmpty ? 'return <Object>[];' : '''
+    ${fields.isEmpty ? 'return <Object$orNull>[];' : '''
```

Comment:
Pardon my intrusion, but I'm writing a Dart code parser, and came across this file in the Flutter distribution.

How is the Dart compiler supposed to decide whether the starting triple single quote (`'''`) on line 306 ends:
* here on line 316 or
* line 321 or
* below on line 328 or
* later on line 342 or
* what I think should be the correct answer, line 345 ?

Reply:
The string interpolation expressions `${...}` completely hide any quotes inside them. Hopefully that's enough to get the right answer :)

Reply:
Not sure how useful this is, but the formal grammar can be found here: https://dart.dev/guides/language/spec